adam-core/src/main/scala/org/bdgenomics/adam/converters/FastaConverter.scala

```diff
@@ -214,6 +214,7 @@ private[converters] class FastaConverter(fragmentLength: Long) extends Serializa
           .setFragmentSequence(bases)
           .setFragmentNumber(index)
           .setFragmentStartPosition(index * fragmentLength)
+          .setFragmentEndPosition(index * fragmentLength + bases.length - 1L)
```

Comment:
It doesn't look like we check this further up in the code, but it might be reasonable to have an assert checking that either:

1. The length of the sequence equals the `fragmentLength` (`bases.length == fragmentLength`), or
2. The fragment is the last fragment of the sequence (`fragmentCount == index`).